core/src/main/scala/org/apache/spark/sql/rapids/tool/util/ToolsPlanGraph.scala

```diff
@@ -187,6 +187,58 @@ case class DBReflectionContainer() {
   }
 }
 
+/**
+ * Extension of SparkPlanGraphNode to handle Photon nodes.
+ * Note:
+ * - photonName and photonDesc are the name and description of the Photon node
+ * - name and desc are the name and description of the equivalent Spark node
+ */
+class PhotonSparkPlanGraphNode(
+    id: Long,
+    val photonName: String,
+    val photonDesc: String,
+    sparkName: String,
+    sparkDesc: String,
```

Comment:
We can revisit this because I think it is redundant to keep this information in node level since the map should be an operator scope rather than node scope. This will also impact memory storage.

Reply:
`sparkName` and `sparkDesc` are passed to `SparkPlanGraphNode` as `node` and `desc` respectively.

This ensures that any call to `node.name` and `node.desc` returns the corresponding Spark name and desc.